numpy/core/src/multiarray/methods.c

```diff
@@ -1133,6 +1133,17 @@ any_array_ufunc_overrides(PyObject *args, PyObject *kwds)
         }
     }
     Py_DECREF(out_kwd_obj);
+    /* check where if it exists */
+    where_obj = _PyDict_GetItemStringWithError(kwds, "where");
```

Comment:
I worry a bit about speed, but `out` is treated very similarly so definitely out of scope here. But maybe @seberg has thoughts...

Reply:
This can be much faster by using a global `where` string rather than `"string"`.  We can't avoid the ~20ns of the dict lookup itself, unfortunately.  Now that we go beyond one key that needs to be fetched from the dict, it could have been nice to do that earlier.

That said, we already have to go through all of the work to build the dict, so that is more work anyway :).

Reply:
So I should go ahead and define a global `where` string right?

Reply:
Yeah, i.e. there: https://github.com/numpy/numpy/blob/main/numpy/core/src/multiarray/multiarraymodule.c#L4901
It probably already is in the umath namespace (which is the same one now).  But could put it there (and at some point we could move the umath ones over maybe).

Reply:
> Now that we go beyond one key that needs to be fetched from the dict, it could have been nice to do that earlier.

Sorry, I don't understand, what could we do earlier?

Reply:
Ignore it.  `__array_function__` passes all objects that need dispatching explicitly (a second time).  Which would be looking nice now that we have 3 things where objects can be found.

Reply:
@seberg, I'm having trouble accessing the new global variable I've created. When I include `multiarraymodule.h` in `methods.c`, I get [linker errors](https://github.com/numpy/numpy/actions/runs/4260484958/jobs/7413700120#step:5:592). Any ideas on where I might be going wrong?

Reply:
@byrdie did you see how the other symbols there were defined used?  You probably forgot the `extern`.

Reply:
Thanks, it ended up being something else, but that was a good guess! I usually forget the extern. Should be all good to go as soon as the CI passes.